dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.35/V2_35_6__Add_Uuid_Generator.sql

```diff
@@ -0,0 +1,12 @@
+
+
+CREATE OR REPLACE FUNCTION gen_random_uuid() RETURNS uuid
+    AS $$
+        SELECT uuid_in(
+            overlay(
+                overlay(md5(random()::text || ':' || clock_timestamp()::text) placing '4' from 13)
+                placing to_hex(floor(random()*(11-8+1) + 8)::int)::text from 17
+            )::cstring
+        )::uuid
+    $$
+    LANGUAGE SQL;
```

Comment:
I'm not 100% sure I understand all parts of this function.
Do you have any links/urls to where you found it or if you made it yourself may write some comments to what is going on.
A test for its uniqueness properties would also be nice.